content-src/components/ASRouterAdmin/ASRouterAdmin.jsx

```diff
@@ -75,13 +98,27 @@ class DiscoveryStreamAdmin extends React.PureComponent {
   }
 
   render() {
+    const {isOptedOut} = this;
+
     const {config, lastUpdated, layout} = this.props.state;
     return (<div>
-      <div className="dsEnabled"><input type="checkbox" checked={config.enabled} onChange={this.onEnableToggle} /> enabled</div>
 
+      <div className="dsEnabled"><input type="checkbox" checked={config.enabled} onChange={this.onEnableToggle} /> enabled
+        {isOptedOut ? (<span className="optOutNote">(Note: User has opted-out. Check this box to reset)</span>) : ""}</div>
+
+      <h3>Endpoint variant</h3>
+      <p>You can also change this manually by changing this pref: <code>browser.newtabpage.activity-stream.discoverystream.config</code></p>
+      <table style={config.enabled ? null : {opacity: 0.5}}><tbody>
+        {Object.keys(LAYOUT_VARIANTS).map(id => (<Row key={id}>
+          <td className="min"><input type="radio" value={id} checked={this.isCurrentVariant(id)} onChange={this.changeEndpointVariant} /></td>
+          <td className="min">{id}</td>
+          <td>{LAYOUT_VARIANTS[id]}</td>
+        </Row>))}
+      </tbody></table>
```

Comment:
Might be useful to have a "full url" textbox to set just the endpoint directly. Currently if it's none of the explicit `LAYOUT_VARIANTS`, nothing is checked. And if the url doesn't match `layout_variant=`, clicking the radio buttons do nothing, e.g., if it's set to `https://gist.githubusercontent.com/punamdahiya/44f78e2f338cbf95f5c28d31faed4639/raw/1c61974ca8ebd02431512d9425031e3a9c8e04f4/test-message-feed.json`

Having the textbox is slightly clearer what's going on with the gist endpoint. But we don't need to spend too much (/ any?) effort here as there's always about:config…

Reply:
I was thinking it might be a good idea to ask security before doing that, but if it's just the end of the URL it's probably fine. I won't do it right now just cause i dont have much time